npm/webpack-dev-server-fresh/src/helpers/sourceRelativeWebpackModules.ts

```diff
@@ -94,30 +98,67 @@ export function sourceRelativeWebpackModules (config: WebpackDevServerConfig) {
 
   let webpackJsonPath: string
 
-  try {
-    webpackJsonPath = require.resolve('webpack/package.json', {
-      paths: [searchRoot],
-    })
-  } catch (e) {
-    if ((e as {code?: string}).code !== 'MODULE_NOT_FOUND') {
+  if (config.framework === 'next') {
+    try {
+      webpackJsonPath = require.resolve('next/dist/compiled/webpack/package.json', {
+        paths: [searchRoot],
+      })
+    } catch (e) {
       throw e
     }
 
-    webpackJsonPath = require.resolve('webpack/package.json', {
-      paths: [
-        require.resolve('@cypress/webpack-batteries-included-preprocessor', {
-          paths: [__dirname],
-        }),
-      ],
-    })
-  }
+    let webpack5 = true
+    const importPath = path.dirname(webpackJsonPath)
+    const webpackModule = require(path.join(importPath, 'webpack.js'))
+
```

Comment:
This entire webpack module file is getting to be really complex really fast, I wonder if we can find some cleaner way to handle this eventually...

Reply:
I agree, @tgriesser might have some good ideas on cleaning this up. I think we should allow the handlers to source the information so we don't have the framework bloat in this file, and `sourceRelativeWebpackModules.ts` could be helper functions that the handlers consume, but not going to refactor in this PR.

Reply:
Agreed, this stuff is all super ugly. It makes me want to have a `next-dev-server` separate from our `webpack-dev-server`, since it's requiring so much custom configuration / patching.

Reply:
Or at the very least its own file / function in `webpack-dev-server`. `webpackConfig = framework === 'nextjs') ? getNextWebpackConfig() : getWebpackConfig()`, and they can share utility functions as needed.

Reply:
The more I look at it the more it feels like something we should decide now rather than later, pull it out in some fashion rather than kick it down the road.

Reply:
I'm going to refactor this! I'm sidetracked getting the dev-servers to work with the built binary but going to return

Reply:
Yeah I think the next stuff should be handled separately than the regular one. 

The first attempt here was intentionally procedural, until we knew what we needed to reuse for next, it didn’t make sense to abstract.

Reply:
Refactored, broke the functions up so they could be reused and altered the handler contract so that each handler is responsible for providing the sourced deps.